### PR TITLE
Ensure proper order of statements in fts_segment_reset test

### PR DESCRIPTION
**Pull request by dkovalev1**
Ensure proper order of statements in the test fts_segment_reset.

The aforementioned isolation test contains statements that are started in
parallel at the same time. Under some circumstances, e.g., when using a Docker
container based on Ubuntu 22.04, these statements are executed in an unexpected
order, causing the test to fail.

This commit adds a fault injector to enforce the order of statements by
introducing a dependency between getting segment master into reset and 
the start of next command.

### CODE REVIEW

File: src/test/isolation2/sql/fts_segment_reset.sql [hunk 1]
```diff
@@ -29,6 +29,11 @@ from gp_segment_configuration where role = 'p' and content = 0;
 from gp_segment_configuration where role = 'p' AND content = 0;
 1&:create table fts_reset_t(a int);
 
+-- Make sure that fault is processed and the following query started after seg0 has actually gone
+-- Of course, sleeping is not the best solution, but wait_until_triggered_fault won't work here 
+-- because seg0 process with faults hash table has gone.
+select pg_sleep(0.5);
```
Comment (RekGRpth) [inline src/test/isolation2/sql/fts_segment_reset.sql]: Can we use something [like](https://github.com/arenadata/gpdb/commit/fd88a67a8f29cf2d537a73c6c04fc492cb06c736)?

Comment (dkovalev1) [reply src/test/isolation2/sql/fts_segment_reset.sql]: I believe cyclic reading of the table with a sleep is not the best solution - it will take much more resources and introduce unneeded complexity. I think wait on injected fault is more suitable here.

**Review by red1452 — APPROVED** [review]
I rewrite the last sentence:
`between getting segment master into reset and
the start of next command` -> `between getting segment master into reset and
starting of  the next command.`

**Comment by KnightMurloc** [comment]
Please describe in more detail (including in the description of PR) what sequence of events leads to the failure of tests and how this patch fixes it.

File: src/test/isolation2/sql/fts_segment_reset.sql [hunk 2]
```diff
@@ -48,7 +59,7 @@ select dbid, role, preferred_role, status from gp_segment_configuration where co
 select gp_inject_fault('postmaster_delay_termination_bg_writer', 'reset', dbid) from gp_segment_configuration where role = 'p' and content = 0;
 
 -- The only table that should have been created successfully
-drop table fts_reset_t3;
+3:drop table fts_reset_t3;
```
Comment (KnightMurloc) [inline src/test/isolation2/sql/fts_segment_reset.sql:62]: Why is this necessary?

Comment (dkovalev1) [reply src/test/isolation2/sql/fts_segment_reset.sql:62]: This is another point if instability. Running statements on a cluster where segments stopping, crashing, reset, etc may lead in broken client session, which is sometime happens for default session in this test. Default session is used in between of operations and can catch segment reset.
Session 3 is not used in the middle of the test and survives till the end, that's why this change introduced.

**Review by KnightMurloc — APPROVED** [review]
